### PR TITLE
Synonym search

### LINKED ISSUE #205: Allow keyword synonyms to return the matching keyword
We store all the synonyms, we just don't search them yet. Return the main keyword for the synonym if it is found. See TOAST for existing implementation.

### PR DESCRIPTION
**Pull request by bgrawi**
This fixes #205 for the backend.

### CODE REVIEW

File: app/controllers/keyword.js [hunk 1]
```diff
@@ -39,19 +39,71 @@ function partialKeywordMatch(req, res, next) {
...
+				.orWhere('external_id', 'LIKE', `%${req.query.substring}%`);
+
+			const synonymQuery = knex.select(synonymFields)
+				.from('synonym')
+				.leftJoin('keyword', 'keyword.id', 'synonym.keyword_id')
+				.where('synonym.name', 'LIKE', `%${req.query.substring}%`);
+
+			// Union the results of the two query 
+			const unionQuery = keywordQuery.union(synonymQuery).as('unionQuery');
+            const finalQuery = knex.select(finalFields).from(unionQuery);
```
Comment (GavinNishizawa) [inline app/controllers/keyword.js]: spacing but #248

File: app/controllers/keyword.js [hunk 2]
```diff
@@ -39,19 +39,71 @@ function partialKeywordMatch(req, res, next) {
...
+			if (needsEcoId) {
+				finalQuery.rightJoin('keyword_mapping', 'unionQuery.external_id', 'keyword_mapping.eco_id');
+			}
+
+			// Make sure the keyword is not obsolete and that it has an external id.
+			finalQuery.whereNot('is_obsolete', 1);
+			finalQuery.whereNotNull('external_id');
+
+			// Order the canonical results first and only show 20.
+			finalQuery.orderBy('synonym', 'desc');
```
Comment (GavinNishizawa) [inline app/controllers/keyword.js]: did we want to do order by synonym then by keyword name?

File: app/controllers/keyword.js (hunk 1 above)
Comment (bgrawi) [reply app/controllers/keyword.js]: yeah, I'll let a linter take care of it ¯\_(ツ)_/¯

File: app/controllers/keyword.js (hunk 2 above)
Comment (bgrawi) [reply app/controllers/keyword.js]: This was the only way I could enforce normal keywords coming up first. No doubt this is a workaround and sorting them by alphabetic order is temporary. I'll wait for feedback from phoenix about better ideas for sorting.

**Pull request merged**